crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -18,6 +20,130 @@ use crate::docstrings::Docstring;
 use crate::registry::Rule;
 use crate::rules::pydocstyle::settings::Convention;
 
+/// ## What it does
+/// Checks for function docstrings that do not include documentation for all
+/// parameters.
+///
+/// ## Why is this bad?
+/// If a function accepts a parameter without documenting it in its docstring,
+/// it can be misleading to users and/or a sign of incomplete documentation or
+/// refactors.
+///
+/// ## Example
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///     """
+///     return distance / time
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent travelling.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///     """
+///     return distance / time
+/// ``` raise FasterThanLightError from exc
+/// ```
+#[violation]
+pub struct DocstringMissingParameter {
+    id: String,
+}
+
+impl Violation for DocstringMissingParameter {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let DocstringMissingParameter { id } = self;
+        format!("Parameter `{id}` missing from docstring")
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        let DocstringMissingParameter { id } = self;
+        Some(format!("Add `{id}` to the docstring"))
+    }
+}
+
+/// ## What it does
+/// Checks for function docstrings that include parameters which are not
+/// in the function signature.
+///
+/// ## Why is this bad?
+/// If a docstring documents a parameter which is not in the function signature,
+/// it can be misleading to users and/or a sign of incomplete documentation or
+/// refactors.
+///
+/// ## Example
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent traveling.
+///         acceleration: Rate of change of speed.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///     """
+///     return distance / time
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent traveling.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///     """
+///     return distance / time
+/// ```
+#[violation]
+pub struct DocstringExtraneousParameter {
+    ids: Vec<String>,
```

Comment:
I like grouping extraneous parameters to reduce the total number of parameters. It only feels a bit inconsistent that we don't group missing parameters the same way. Is it an intentional decision to only group extraneous parameters?